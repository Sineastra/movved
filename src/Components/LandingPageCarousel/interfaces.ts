export interface slideIntF {
	image: string,
	link: string
}