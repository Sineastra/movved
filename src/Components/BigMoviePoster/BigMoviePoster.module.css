

/*# sourceMappingURL=BigMoviePoster.module.css.map */
