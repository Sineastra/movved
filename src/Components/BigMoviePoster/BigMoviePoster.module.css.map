{"version":3,"sourceRoot":"","sources":["BigMoviePoster.module.scss","../../_scssPartials/mixins/_mixins.scss"],"names":[],"mappings":"AAEA;EACI;EACA;EACA;;AAEA;EACI;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EACA;EACA;EACA;EACA;;AAGJ;EACI;EACA;EACA;EACA;EC3BJ;EACA,gBD2ByB;EC1BzB,iBD0BiC;ECzBjC,aAJgE;EAKhE,eALwF;;AD+BpF;EACI;EACA;;AAGJ;EACI","file":"BigMoviePoster.module.css"}