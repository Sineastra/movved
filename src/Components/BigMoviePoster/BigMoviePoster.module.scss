@import "../../_scssPartials/mixins/mixins";

.posterWrapper {
    width: 100vw;
    height: 300px;
    position: relative;

    & > img {
        width: 100vw;
        height: 300px;
        object-fit: cover;
    }

    &::after {
        left: 0;
        bottom: 0;
        z-index: 1;
        content: "";
        width: 100vw;
        height: 100px;
        position: absolute;
        background: linear-gradient(180deg, rgba(20, 20, 20, 0), rgba(20, 20, 20, .8) 51.04%, rgba(28, 28, 28, 1) 72.92%, var(--theme-main-color_2));
    }

    & .headingsWrapper {
        bottom: 0;
        z-index: 2;
        left: 10px;
        position: absolute;
        @include displayFlex(column, center);

        & > h2, h4 {
            margin: 0 0 5px;
            font-family: 'Comfortaa', cursive;
        }

        & > h4 {
            color: var(--theme-subtext-color);
        }
    }

}