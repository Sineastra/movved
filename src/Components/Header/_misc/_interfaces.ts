import { signedUserInfoInterface } from "../../../_misc/interfaces"
import React from "react"


export interface desktopPropsInterface {
	loggedUser: signedUserInfoInterface | null,
	className?: string,
}

export interface asideNavPropsInterface {
	loggedUser: signedUserInfoInterface | null,
	className?: string,
}

export interface mobilePropsInterface {
	toggleAsideNav: (event: React.MouseEvent) => void,
	className?: string
}