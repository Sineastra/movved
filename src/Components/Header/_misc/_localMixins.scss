@import "../../../_scssPartials/mixins/mixins";

@mixin aside-nav-link {
    height: 25%;
    text-align: start;
    text-decoration: none;
    @include centeredFlex(row);
    color: var(--theme-text-color);

    &:hover {
        color: var(--theme-secondary-color)
    }
}