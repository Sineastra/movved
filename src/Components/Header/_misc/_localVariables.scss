$aside-nav-height: 300px;
$header-height: 65px;