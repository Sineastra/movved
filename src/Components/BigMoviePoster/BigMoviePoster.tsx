import styles from "./BigMoviePoster.module.scss"
import { movie } from "../../_misc/interfaces"


interface propsIntF {
	movie: movie
}
const BigMoviePosterTest = ({ movie }: propsIntF) => {
	return (
		<section className={ styles.posterWrapper }>
			<img src={ movie.bigPoster } alt="big-poster"/>
			<div className={ styles.headingsWrapper }>
				<h2>{ movie.title }</h2>
				<h4>{ movie.engTitle }</h4>
			</div>
		</section>
	)
}

export default BigMoviePosterTest