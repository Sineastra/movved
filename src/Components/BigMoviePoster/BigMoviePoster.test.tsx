import renderer from "react-test-renderer"
import BigMoviePoster from "./BigMoviePoster"
import { movieInterface } from "../../_misc/interfaces"


const mockedMovie: movieInterface = {
	_id: '0',
	title: 'a',
	smallPoster: 'b',
	bigPoster: 'c',
	isSubbed: true,
	isDubbed: true,
	year: 0,
	watchLink: 'd',
	engTitle: 'e',
	genres: ['f'],
	trailerLink: 'a',
	comments: [{
		_id: 'a',
		user: {
			_id: '1',
			name: 'a',
			email: 'a',
			password: 'a',
			profilePic: 'a',
		}, comment: 'comment',
	}],
	actors: ['a', 'b'],
	director: 'a',
}

describe("---> Snapshot test of BigMoviePoster", () => {
	it("should render properly", () => {
		const tree = renderer.create(<BigMoviePoster movie={ mockedMovie }/>).toJSON()

		expect(tree).toMatchSnapshot()
	})
})