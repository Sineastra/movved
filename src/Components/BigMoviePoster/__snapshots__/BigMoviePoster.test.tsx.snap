// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`---> Snapshot test of BigMoviePoster should render properly 1`] = `
<section
  className="posterWrapper"
>
  <img
    alt="big-poster"
    src="c"
  />
  <div
    className="headingsWrapper"
  >
    <h2>
      a
    </h2>
    <h4>
      e
    </h4>
  </div>
</section>
`;
